libcpuid/cpuid_main.c

```diff
@@ -682,8 +682,77 @@ int cpu_ident_internal(struct cpu_raw_data_t* raw, struct cpu_id_t* data, struct
 
 int cpu_identify(struct cpu_raw_data_t* raw, struct cpu_id_t* data)
 {
+	int r;
 	struct internal_id_info_t throwaway;
-	return cpu_ident_internal(raw, data, &throwaway);
+	r = cpu_ident_internal(raw, data, &throwaway);
+	data->affinity_mask = (1 << data->num_logical_cpus) - 1;
+	return r;
+}
+
+int cpu_identify_all(struct cpu_raw_data_array_t *raw_array, struct system_id_t* system)
+{
+	int cur_error = set_error(ERR_OK);
+	int ret_error = set_error(ERR_OK);	bool is_new_cpu_type;
+	uint8_t logical_cpu;
+	uint8_t cpu_type_index = 0;
+	int32_t core_previous_id = -1;
+	int32_t num_cores = 1;
+	int32_t num_logical_cpus = 1;
+	uint32_t affinity_mask = 0x00000001;
```

Comment:
The affinity mask should be uint64_t if we want to support more than 32-thread CPUs.

Reply:
Fixed in d204dcfd427e050df35a27440bfc98d8c6c40b5e.

Reply:
Well, I will follow your comment below. You are right, even a uint64_t is still a limitation.

Reply:
Done in 3a6370474468128918cab560fd26be178e094370.